### PR TITLE
GH-84559: Deprecate fork being the multiprocessing default.

### PR DESCRIPTION
**Pull request by gpshead**
This starts the process. Users who don't specify their own start method and use the default on platforms where it is 'fork' will see a DeprecationWarning upon multiprocessing.Pool() construction or upon multiprocessing.Process.start().

* Issue: gh-84559

### CODE REVIEW

**Comment by gpshead** [comment]
This code could be a lot cleaner if https://github.com/python/cpython/issues/39615 were implemented...

File: Doc/library/concurrent.futures.rst [hunk 1]
```diff
@@ -284,6 +284,13 @@ to a :class:`ProcessPoolExecutor` will result in deadlock.
       The *max_tasks_per_child* argument was added to allow users to
       control the lifetime of workers in the pool.
 
+   .. versionchanged:: 3.12
+      The implcit use of the :mod:`multiprocessing` *fork* start method as a
```
Comment (FFY00) [inline Doc/library/concurrent.futures.rst]:
```suggestion
      The implicit use of the :mod:`multiprocessing` *fork* start method as a
```

**Review by gvanrossum — COMMENTED** [review]
Changes to test_asyncio LGTM. I hope I won't have to review the rest, I know basically nothing about multiprocessing. :-(

**Review by kumaraditya303 — COMMENTED** [review]
I approve the `asyncio` change.

File: Lib/multiprocessing/context.py [hunk 2]
```diff
@@ -53,7 +56,9 @@ def Manager(self):
         can be used to create shared objects.
         '''
         from .managers import SyncManager
-        m = SyncManager(ctx=self.get_context())
+        ctx = self.get_context()
+        m = SyncManager(ctx=ctx)
+        proc_class = ctx.Process
```
Comment (Yhg1s) [inline Lib/multiprocessing/context.py]: proc_class isn't used?

**Review by Yhg1s — APPROVED** [review]

File: Lib/multiprocessing/context.py (hunk 2 above)
Comment (gpshead) [reply Lib/multiprocessing/context.py]: leftover from an earlier iteration of the code.  removed.

**Comment by hugovk** [comment]
Please could you add this to What's New, under [Pending Removal in Python 3.14](https://docs.python.org/3.12/whatsnew/3.12.html#pending-removal-in-python-3-14)?

**Comment by gpshead** [comment]
> Please could you add this to What's New, under [Pending Removal in Python 3.14](https://docs.python.org/3.12/whatsnew/3.12.html#pending-removal-in-python-3-14)?

Ahha, I knew we had a list of upcoming things somewhere. Good idea, done! What's New text added.

**Pull request merged**